File: clojure/gigasecond/gigasecond.clj
(ns gigasecond
  (:import [java.util Calendar GregorianCalendar]))

(defn from [year month day]
  (let [cal (GregorianCalendar. year (dec month) day)
        billion 1000000000]
    (.add cal Calendar/SECOND billion)
    [(.get cal Calendar/YEAR)
     (inc (.get cal Calendar/MONTH))
     (.get cal Calendar/DAY_OF_MONTH)]))

File: erlang/gigasecond/gigasecond.erl
-module(gigasecond).
-export([from/1]).

-define(BILLION, 1000000000).

from(Date = {_Y, _M, _D}) ->
    from({Date, {0, 0, 0}});
from({Date, Time}) ->
    Seconds = calendar:datetime_to_gregorian_seconds({Date, Time}),
    calendar:gregorian_seconds_to_datetime(Seconds + ?BILLION).
